### DISCUSSION TITLE
第2节课作业(7班)

### DISCUSSION THREAD

**Discussion by JimmyMa99** [InternLM/Tutorial#111] [作业]
第2节课作业(7班)

**Comment by wuhaostudio** [comment]
作业打卡
<img width="1080" alt="微信图片_20240106162802" src="https://github.com/InternLM/tutorial/assets/143194075/6fa14d77-be1a-49ad-819a-f11067a09c22">
<img width="1056" alt="微信图片_20240106162811" src="https://github.com/InternLM/tutorial/assets/143194075/47e92196-d40d-47b6-92ca-d942a734e235">
<img width="1043" alt="微信图片_20240106162817" src="https://github.com/InternLM/tutorial/assets/143194075/c7cb4618-9e4f-41e9-a6e9-7180a3467d98">
<img width="807" alt="微信图片_20240106162821" src="https://github.com/InternLM/tutorial/assets/143194075/268898f1-e976-461c-9ed1-cc0d2746812f">
<img width="1038" alt="微信图片_20240106162825" src="https://github.com/InternLM/tutorial/assets/143194075/b5f4d8eb-6928-4a37-a9ee-810b98416f14">

**Comment by leochiang** [comment]
基础作业：
![1](https://github.com/InternLM/tutorial/assets/7628117/5d1fff5f-95ed-44b4-b9fb-9fcdc26cf1d6)
![2](https://github.com/InternLM/tutorial/assets/7628117/42abf006-eb7e-4ed9-8920-d3ffdf7a06a7)
进阶作业
![3](https://github.com/InternLM/tutorial/assets/7628117/1e4f74b6-ae7c-4afd-8129-80a324c096a8)

**Comment by createup2023** [comment]
链接：https://www.yuque.com/creatup/ynrsde/itokdeeuirsvq1ue?singleDoc# 《作业》
● 使用 InternLM-Chat-7B 模型生成 300 字的小故事（需截图）。
 
![image](https://github.com/InternLM/tutorial/assets/128460366/149fcae9-9b46-4dbb-a8d1-7fe8f55a6f29)

● 熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地（需截图下载过程）。
![image](https://github.com/InternLM/tutorial/assets/128460366/27b27751-68dc-4938-b2f0-8b725656be94)

上面的会超时，使用Modelscope下载：
![image](https://github.com/InternLM/tutorial/assets/128460366/b21ff24b-259d-4a28-aa9f-ba5566d21d7b)

进阶作业（可选做）
● 完成浦语·灵笔的图文理解及创作部署（需截图）
python3 examples/web_demo.py  \
--folder /root/model/Shanghai_AI_Laboratory/internlm-xcomposer-7b \
--num_gpus 1 \
--port 6006
![image](https://github.com/InternLM/tutorial/assets/128460366/ab6e7670-6298-4814-9299-c5635b87eb4d)

● 完成 Lagent 工具调用 Demo 创作部署（需截图）
![image](https://github.com/InternLM/tutorial/assets/128460366/b818c1aa-506b-452a-b370-78eeb2795269)

**Comment by PommesPeter** [comment]
## 第二次课程作业

1. 300 字小故事
![image](https://github.com/InternLM/tutorial/assets/54879512/2dea2fcb-ff5b-4bdb-90da-a15daf944903)

2. huggingface_hub 使用
更换 HF_ENDPOINT
```bash
export HF_ENDPOINT=https://hf-mirror.com
```
下载
```bash
huggingface-cli download --resume-download --local-dir-use-symlinks False internlm/internlm-chat-7b
```
![image](https://github.com/InternLM/tutorial/assets/54879512/ea407226-47dd-4be2-bd50-a7e0eaf473dd)
![image](https://github.com/InternLM/tutorial/assets/54879512/acfd982e-9a39-4aca-b6b5-d199c46e1277)

3. 浦语·灵笔的图文理解及创作部署
图文并茂生成
![image](https://github.com/InternLM/tutorial/assets/54879512/98831ac7-6184-4f6b-b663-b2b2238e8f02)

多模态理解
![image](https://github.com/InternLM/tutorial/assets/54879512/e8187fdf-194d-4a87-a19f-421277519df7)


4. Lagent 工具调用 Demo 创作部署
![image](https://github.com/InternLM/tutorial/assets/54879512/f1015b7b-db43-44f4-a80a-c48ca95f44d8)

**Comment by YangXuanyi** [comment]
## 第二节基础作业
### 1 生成300字小故事
![作业1-生成故事](https://github.com/InternLM/tutorial/assets/83216339/ee0269a0-2bd1-4972-b6e6-97fcb4f6d22b)

### 2 下载json文件
![10 2](https://github.com/InternLM/tutorial/assets/83216339/1b4407c6-68d1-4266-ab1d-0ff66a88d49d)

### 疑惑
在下载json文件时，我发现modelscope中的snapshot_download下载函数虽然能够快速下载，但是没有找到指定下载某一文件的选项。我跳转到函数里也没有看到这一参数，请问如何使用modelscope中的snapshot_download实现特定文件下载？

**Comment by songqiang321** [comment]
第二讲作业：
https://www.xiaohongshu.com/explore/659abdaf000000001a002344

**Comment by hui1feng** [comment]
第二次作业：https://github.com/hui1feng/-Demo/blob/main/%E4%BD%9C%E4%B8%9A.ipynb

**Comment by SaaRaaS-1300** [comment]
第2堂课的笔记及作业链接：github-> https://github.com/SaaRaaS-1300/InternLM_openNotebook/blob/main/Lesson-2/Lesson-2-Notebook.md

**Comment by shufanwu** [comment]
第二次课作业：https://zhuanlan.zhihu.com/p/676468368

**Comment by anneeeee** [comment]
完成了全部基础作业和进阶作业

1. chat对话 - CLI调用
![image](https://github.com/InternLM/tutorial/assets/16276686/76416dfe-c5d5-4ad4-8d20-fcf25fd7af57)

2. chat对话 - web调用
<img width="1000" alt="image" src="https://github.com/InternLM/tutorial/assets/16276686/47cf6e44-d5cf-4cce-8fdc-5143b2115c8f">

3. Huggingface下载
![image](https://github.com/InternLM/tutorial/assets/16276686/250fe884-a57c-4061-8a5b-11ae78e5838b)

4. 又见敦煌
部署：
![image](https://github.com/InternLM/tutorial/assets/16276686/d81f161a-1028-4b1c-b7c2-ba5cfc79ab3b)

生成图片：
![image](https://github.com/InternLM/tutorial/assets/16276686/b906a074-6563-48bd-9e8a-d14434305f3c)

结果：
![image](https://github.com/InternLM/tutorial/assets/16276686/633cbbd7-699d-4fa2-b29e-b068829ec4a4)


6. 图文理解
![image](https://github.com/InternLM/tutorial/assets/16276686/a5bc18a7-283f-4ac8-8b9d-d38583a7e8a8)

8. lagent
![image](https://github.com/InternLM/tutorial/assets/16276686/869dc697-bf6c-40eb-92a9-1be05b4ba33d)

**Comment by Rogue-Trader-zzy** [comment]
第二次打卡：https://zhuanlan.zhihu.com/p/676951777

**Comment by lifelinezhang** [comment]
https://zhuanlan.zhihu.com/p/676476791

**Comment by snowhou** [comment]
第二课作业打卡：https://zhuanlan.zhihu.com/p/677131304?

**Comment by abelezm** [comment]
第二课笔记和作业打卡：https://blog.csdn.net/weixin_38043453/article/details/135437893

**Comment by codeview126** [comment]
使用 InternLM-Chat-7B 模型生成 300 字的小故事：
![20240107_01](https://github.com/InternLM/tutorial/assets/127104394/6b1a58d9-b64b-4130-ae56-fed015b82b4c)

完成浦语·灵笔的图文理解及创作部署（需截图）
![20240107_04](https://github.com/InternLM/tutorial/assets/127104394/1e4e1ac4-d66b-4aeb-940e-04c826e4ddcd)


熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地

**Comment by Simon-lyz** [comment]
## InternLM lesson-2 web_demo部署 + Agent Prompt与运行逻辑研读
https://g4w90egc30.feishu.cn/docx/KIw4ddOrQoIeWCxJMjbciXb8nbf?from=from_copylink

## Lagent源码阅读 ReAct、Rewoo、AutoGPT
https://g4w90egc30.feishu.cn/docx/QhDLdN5BUou8sAxXJ2rccOotntd?from=from_copylink
